File: FinneJobber.Models/Job.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FinneJobber.Models
{
    public class Job
    {
        public int Id { get; set; }

        [Required]
        [DataType(DataType.Date)]
        [Display(Name = "Date")]
        public DateTime Date { get; set; }

        [Required]
        [DataType(DataType.Time)]
        [DisplayFormat(ApplyFormatInEditMode = true, DataFormatString = "{0:HH:mm}")]
        public DateTime Time { get; set; }
        public string Description { get; set; }
        [Required]
        public double Budget { get; set; }
        [Required]
        public string JobLocation { get; set; }
        [Required]
        public int CategoryId { get; set; }
        [ForeignKey("CategoryId")]
        [ValidateNever]
        public Category Category { get; set; }
    }
}

File: FinneJobberWeb/wwwroot/js/job.js
var dataTable;

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": {
            "url": "/Admin/Job/GetAll"
        },
        "columns": [
            { "data": "category.name", "width": "15%" },
            { "data": "description", "width": "15%" },
            { "data": "jobLocation", "width": "15%" },
            { "data": "date", "width": "15%" },
            { "data": "time", "width": "15%" },
            { "data": "budget", "width": "15%" },
        ]
    });
} 